### PR TITLE
Delete artifact files after sucessful transaction

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

**Review by gerrod3 — APPROVED** [review]
Nice fix!

File: pulpcore/app/models/content.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def delete(self, *args, **kwargs):
             kwargs (dict): dictionary of keyword arguments to pass to Model.delete()
         """
         super().delete(*args, **kwargs)
-        self.file.delete(save=False)
+        transaction.on_commit(partial(self.file.delete, save=False))
```
Comment (dralley) [inline pulpcore/app/models/content.py:165]: Let's leave a comment about why this is being done

Comment (dralley) [reply pulpcore/app/models/content.py:165]: Can we test the basic function here manually before merging?  e.g. create an artifact, delete the artifact record without deleting the file somehow, then use a reference to the filefield to delete the remaining file without the record being present.

Just to make sure there's no unexpected behavior.  Also should we be careful of `self.file` being nulled out or invalidated behind our backs?

Comment (dralley) [reply pulpcore/app/models/content.py:166]: >having a nullified file field (and such a thing cannot come from the database [1])

I recall seeing it use an empty string in some cases, though this was an experiment I did more than a year ago so I can't remember the details.

**Review by dralley — APPROVED** [review]

**Pull request merged**